components/sponsor/NftSubmissionForm.vue

```diff
@@ -0,0 +1,552 @@
+    set(email, submission.email || '');
+    // Clear image fields as we can't load existing images
+    set(imageFile, null);
+    set(imagePreview, '');
+    // When editing, assume NFT ownership is valid (they already submitted it)
+    set(isNftOwnerValid, true);
+    // Check NFT metadata for the loaded submission
+    checkNftMetadata();
+  }
+}, { immediate: true });
+
+const route = useRoute();
+
+onMounted(() => {
+  const tokenIdParam = route.query.tokenId as string;
+  if (tokenIdParam && !props.editingSubmission) {
+    set(tokenId, tokenIdParam);
+  }
+});
+</script>
+
+<template>
+  <RuiCard class="!p-4">
+    <form
+      class="flex flex-col gap-6"
+      @submit.prevent="handleSubmit()"
+    >
+      <div class="flex items-start gap-2">
+        <RuiAutoComplete
+          v-model="tokenId"
+          :label="t('sponsor.submit_name.token_id_label')"
+          :hint="t('sponsor.submit_name.token_id_hint')"
+          :error-messages="toMessages(v$.tokenId)"
+          :disabled="isSubmitting"
+          :options="nftIdOptions"
+          clearable
+          custom-value
+          auto-select-first
+          variant="outlined"
+          color="primary"
+          class="flex-1"
+        />
+        <RuiButton
+          :disabled="!tokenId || isSubmitting || !isConnected"
+          :loading="isCheckingNft"
+          variant="outlined"
+          color="primary"
+          class="h-14"
+          @click="checkNftMetadata()"
+        >
+          {{ t('sponsor.submit_name.check') }}
+        </RuiButton>
+      </div>
+
+      <RuiAlert
+        v-if="nftCheckError"
+        type="error"
+        class="-mt-2"
+      >
+        {{ nftCheckError }}
+      </RuiAlert>
+      <RuiAlert
+        v-else-if="nftTier"
+        type="info"
+        class="-mt-2"
+      >
+        <i18n-t
+          keypath="sponsor.submit_name.nft_info"
+          tag="span"
+        >
+          <template #tokenId>
+            {{ tokenId }}
+          </template>
+          <template #tier>
+            <strong class="uppercase">{{ nftTier }} tier</strong>
+          </template>
+          <template #releaseName>
+            <strong>{{ nftReleaseName }}</strong>
+          </template>
+        </i18n-t>
+      </RuiAlert>
+
+      <RuiTextField
+        v-model="displayName"
+        :label="t('sponsor.submit_name.name_label')"
+        :hint="t('sponsor.submit_name.name_hint')"
+        :error-messages="toMessages(v$.displayName)"
+        :disabled="isSubmitting"
+        variant="outlined"
+        color="primary"
+      />
+
+      <div v-if="nftTier === 'silver' || nftTier === 'gold'">
+        <label class="block text-sm font-medium mb-2">
+          {{ t('sponsor.submit_name.image_label') }}
+          <span class="text-rui-text-secondary">({{ t('sponsor.submit_name.optional') }})</span>
+        </label>
+
+        <div
+          v-if="!imagePreview"
+          class="relative mt-3"
+        >
+          <input
```

Comment:
not for now, but we should probably create a unified upload control and move it to the library

Reply:
https://github.com/rotki/ui-library/issues/388